### PR TITLE
Task1 4 Slava.Telyatko

### PR DESCRIPTION
**Pull request by SlavaTel**
Screenshot_1
Screenshot_2

### CODE REVIEW

**Comment by SlavaTel** [comment]
https://travis-ci.org/github/SlavaTel/js-assignments/jobs/704955265

**Comment by AlexMokin** [comment]
Not all tests pass.

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by SlavaTel** [comment]
https://travis-ci.org/github/Happy-Love/js-assignments/builds/703587438

**Comment by AlexMokin** [comment]
travis link must point to the same repository as the repository form which you submited a PR

**Comment by SlavaTel** [comment]
https://travis-ci.org/github/SlavaTel/js-assignments

**Comment by FedorovD** [comment]
https://travis-ci.org/github/SlavaTel/js-assignments/builds/707002099

File: task/02-numbers-tasks.js [hunk 1]
```diff
@@ -202,7 +202,11 @@ function roundToPowerOfTen(num, pow) {
  *   17 => true
  */
 function isPrime(n) {
-    throw new Error('Not implemented');
+
+    for(let i = 2; i <= Math.sqrt(n); i++)
+        if(n % i === 0 )
+            return false
+    return true
```
Comment (FedorovD) [inline task/02-numbers-tasks.js]: don't forget to use curly braces, it improves the code readability
https://javascript.info/coding-style

File: task/02-numbers-tasks.js [hunk 2]
```diff
@@ -221,7 +225,7 @@ function isPrime(n) {
  *   toNumber(new Number(42), 0) => 42
  */
 function toNumber(value, def) {
-    throw new Error('Not implemented');
+    return + value ? + value : def;
```
Comment (FedorovD) [inline task/02-numbers-tasks.js]: it's not correct if value = '0', need check that value is a `number`

**Comment by FedorovD** [comment]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by FedorovD** [comment]
You still have not fixed `isPrime` and `toNumber` tasks

**Comment by SlavaTel** [comment]
I fixed isPrime. Why don't you accept my solution to the toNumber task. Every 'value' have to conver to number. 

image

**Comment by FedorovD** [comment]
> I fixed isPrime. Why don't you accept my solution to the toNumber task. Every 'value' have to conver to number.

because 0 is a {number} too, but in this case will be returned def value, you need to check that value is any {number} and return it

File: task/02-numbers-tasks.js [hunk 3]
```diff
@@ -228,7 +228,7 @@ function isPrime(n) {
  *   toNumber(new Number(42), 0) => 42
  */
 function toNumber(value, def) {
-    return Number(value) || def;
+    return (value instanceof Number || +value === parseInt(value)) ? value : def;
```
Comment (FedorovD) [inline task/02-numbers-tasks.js]: 1st: should be return {number} type
2nd: see https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/NaN

**Comment by FedorovD** [comment]
**try to parse "0" or  0 value through your `toNumber` function, 0 is  a number so it should return  this value, not `def` as in your solution.**
click "ready for review" when it will be finished

**Comment by FedorovD** [comment]
Accepted.

**Pull request closed without merging**